From aaf897f358de711f37506f6549a3c5d3a00dedce Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:24:26 -0400
Subject: [tools] perf/trace: Remove duplicate mmap entry in syscall_fmts array

Message-id: <1408462094-14194-98-git-send-email-jolsa@redhat.com>
Patchwork-id: 88104
O-Subject: [PATCH RHEL7.1 BZ1131394 097/325] perf trace: Remove duplicate mmap entry in syscall_fmts array
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 770480592084d2c114adedfbe6740e345aaf2279
Author: David Ahern <dsahern@gmail.com>
Date: Wed Sep 4 12:37:44 2013 -0600

description
===========
Entries in syscall_fmts need to be in alphabetical order, and the
duplicate entry breaks bsearch on new entries around this duplicate
entry.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-trace.c b/tools/perf/builtin-trace.c
index a846688..38782a7 100644
--- a/tools/perf/builtin-trace.c
+++ b/tools/perf/builtin-trace.c
@@ -461,7 +461,6 @@ static struct syscall_fmt {
  { .name     = "arch_prctl", .errmsg = true, .alias = "prctl", },
  { .name     = "brk",     .hexret = true,
    .arg_scnprintf = { [0] = SCA_HEX, /* brk */ }, },
- { .name     = "mmap",     .hexret = true, },
  { .name     = "connect",    .errmsg = true, },
  { .name     = "fcntl",     .errmsg = true,
    .arg_scnprintf = { [1] = SCA_STRARRAY, /* cmd */ },
-- 
1.7.1